sqlmesh/core/engine_adapter/base.py

```diff
@@ -575,45 +602,20 @@ def _escape_json(cls, value: Query | str) -> Query | str:
             )
         return value
 
-    def _insert_append_query(
-        self,
-        table_name: TableName,
-        query: Query,
-        columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
-    ) -> None:
-        column_names = list(columns_to_types or [])
-        self.execute(exp.insert(query, table_name, columns=column_names))
-
-    def _insert_append_pandas_df(
+    def _insert_append_source_queries(
         self,
         table_name: TableName,
-        df: pd.DataFrame,
+        source_queries: t.List[SourceQuery],
         columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
         contains_json: bool = False,
     ) -> None:
-        connection = self._connection_pool.get()
-        table = exp.to_table(table_name)
-
-        sqlalchemy = optional_import("sqlalchemy")
-        # pandas to_sql doesn't support insert overwrite, it only supports deleting the table or appending
-        if sqlalchemy and isinstance(connection, sqlalchemy.engine.Connectable):
-            df.to_sql(
-                table.sql(dialect=self.dialect),
-                connection,
-                if_exists="append",
-                index=False,
-                chunksize=self.DEFAULT_BATCH_SIZE,
-                method="multi",
-            )
-        else:
-            column_names = list(columns_to_types or [])
-            with self.transaction():
-                for i, expression in enumerate(
-                    self._pandas_to_sql(
-                        df, columns_to_types, self.DEFAULT_BATCH_SIZE, contains_json=contains_json
-                    )
-                ):
-                    self.execute(exp.insert(expression, table_name, columns=column_names))
+        with self.transaction(condition=len(source_queries) > 0):
```

Comment:
just always run the transaction, it's fine, remove this condition stuff

Reply:
I would prefer if we only issues transactions if they are needed.